common/compositor/nativesurface.cpp

```diff
@@ -72,13 +70,7 @@ void NativeSurface::SetNativeFence(int fd) {
 }
 
 void NativeSurface::SetInUse(bool inuse) {
-  if (inuse) {
-    ref_count_ = 3;
-  } else if (ref_count_) {
-    ref_count_--;
-  }
-
-  in_flight_ = false;
+  in_use_ = inuse;
```

Comment:
We will need a spinlock here. Since InUse will be queried in application thread and SetinUse can be called from app thread and kmsfencehandler thread